### PR TITLE
Switch to using neg_binomial_2_log

### PR DESCRIPTION
**Pull request by seabbs**
This PR attempts to stabilise and optimise the calculation of expected observations by increasing the proportion of the calculation performed on the log scale. This results in reduced computation time with the majority of this coming from switching to using the `neg_binomial_2_log` family of functions (over their natural scale counterparts).

I have tested this on the readme and found a speed-up of approximately ~ 15%. I haven't observed any negative side effects apart from a slight reduction in clarity. There is potentially more to be done here and I wonder if the repeated logit (and inverse logit) transforms could in some way be optimised.

### CODE REVIEW

File: inst/stan/functions/hazard.stan [hunk 1]
```diff
@@ -20,13 +20,12 @@ vector cumulative_converse_hazard(vector h) {
...
-  ch = exp(ch);
   return(ch);
 }
 
-vector hazard_to_prob(vector h) {
+vector hazard_to_log_prob(vector h) {
   int l = num_elements(h);
   vector[l] p;
-  p[1:l] = h[1:l] .* cumulative_converse_hazard(h[1:l]);
+  p[1:l] = log(h[1:l]) + cumulative_converse_hazard(h[1:l]);
```
Comment (adrian-lison) [inline inst/stan/functions/hazard.stan:29]: I was only wondering whether we should also rename the `cumulative_converse_hazard` to `cumulative_converse_log_hazard` to indicate that it converts the hazard from unit to log scale.

**Review by adrian-lison — COMMENTED** [review]
This is awesome! All looking great, one minor comment.
I might go and steal this for my dev model, I could imagine this being even more impactful if you have a renewal process (as the sampler explores large R values) ;)

File: inst/stan/functions/hazard.stan (hunk 1 above)
Comment (seabbs) [reply inst/stan/functions/hazard.stan:29]: yeah, that is a good point.

Comment (seabbs) [reply inst/stan/functions/hazard.stan:29]: I'll make another PR.

Comment (seabbs) [reply inst/stan/functions/hazard.stan:29]: PR: #69

**Pull request merged**